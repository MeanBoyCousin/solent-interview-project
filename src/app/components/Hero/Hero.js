import React from 'react'

import { HeroWrapper } from './Hero.styled'
import { InputSearch } from '../InputSearch/InputSearch'

const Hero = ({ setQuery }) => {
    return (
        <HeroWrapper role="banner">
            <h1>Find a member of staff</h1>
            <p>Use the search box to get started</p>
            <InputSearch setQuery={setQuery} />
        </HeroWrapper>
    )
}

export { Hero }
