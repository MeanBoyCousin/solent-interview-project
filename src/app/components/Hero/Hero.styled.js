import styled from 'styled-components'

const HeroWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    /* Set black background as fall back if image fails to load. */
    background-color: black;
    background-image: url('/static/images/background.jpg');
    background-position: 35% 0%;
    background-size: cover;
    color: white;
    text-align: center;

    h1 {
        font-size: 2.5rem;
        padding: 15% 5% 10% 5%;
    }

    p {
        padding-bottom: 10%;
    }

    @media (min-width: 992px) {
        text-align: left;

        h1,
        p {
            width: 50%;
        }

        h1 {
            font-size: 4rem;
            padding: 15% 5% 2.5% 5%;
        }

        p {
            font-size: 1.5rem;
            padding-bottom: 5%;
        }
    }
`

export { HeroWrapper }
