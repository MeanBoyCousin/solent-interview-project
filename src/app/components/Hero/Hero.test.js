import React from 'react'
import 'dotenv/config'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { Hero } from './Hero'

afterEach(cleanup)

describe('Hero component', () => {
    test('should render with correct elements', () => {
        const { asFragment } = render(<Hero />)
        expect(asFragment()).toMatchSnapshot()
    })
})
