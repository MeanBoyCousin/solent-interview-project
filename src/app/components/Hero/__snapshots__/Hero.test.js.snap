// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render with correct elements 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa duMbDV"
    role="banner"
  >
    <h1>
      Find a member of staff
    </h1>
    <p>
      Use the search box to get started
    </p>
    <div
      class="sc-bwzfXH cAdroO"
    >
      <img
        alt="A search icon indicating that this input box is for searching the page content"
        src="/static/icons/search.svg"
      />
      <input
        aria-label="Search for a member of staff"
        placeholder="Start typing..."
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
